shared/js/smart-screen/scrollable.js

```diff
@@ -378,10 +382,18 @@
 
       var newFocus = this.getItemFromNode(this.nodes[newFocusIdx]);
       this.nodes = newNodes;
-      if (!this.isHovering) {
+      if (!this.isHovering && newFocus) {
+        // When currently no node is hovering over a folder and the scrollable
+        // is not empty after removing nodes,
+        // reset node positions and the next focus item.
         this._setNodesPosition();
         this.spatialNavigator.focus(newFocus);
-      } else {
+      } else if (this.isHovering) {
+        // When the removed node was hovering over a folder, after it's removed,
+        // reset the idx of the remaining nodes, set the focus to the hovered
+        // folder, fire the event to notify the remaining move to folder actions
+        // and unhover silently without firing an event.
```

Comment:
You should mention something like "In case of hovering, we only remove the hovering item".
Also, although we don't have this case for now, if we try to remove elements other than hovering item, this if clause would cause unexpected behavior.
If you would like to keep this condition here, you need to change the condition to ```this.isHovering && indices.length === 1 && this.hoveringItem === this.getItemFromNode(this.node[indices[0]])```